account_fx_spot/models/account_fx_spot.py

```diff
@@ -0,0 +1,354 @@
+                        in_residual += line.amount_residual_currency if \
+                            line.currency_id else line.amount_residual
+                    else:
+                        from_currency = (
+                            (line.currency_id and line.currency_id.
+                             with_context(date=line.date)) or
+                            line.company_id.currency_id.
+                            with_context(date=line.date))
+                        in_residual += from_currency.compute(
+                            line.amount_residual,
+                            rec.in_currency_id)
+                elif line.account_id.internal_type == "payable":
+                    out_residual_company += line.amount_residual
+                    if line.currency_id == rec.out_currency_id:
+                        out_residual += line.amount_residual_currency if \
+                            line.currency_id else line.amount_residual
+                    else:
+                        from_currency = (
+                            (line.currency_id and line.currency_id.
+                             with_context(date=line.date)) or
+                            line.company_id.currency_id.
+                            with_context(date=line.date))
+                        out_residual += from_currency.compute(
+                            line.amount_residual,
+                            rec.out_currency_id)
+            rec.residual_company_in = abs(in_residual_company)
+            rec.residual_in = abs(in_residual)
+            rec.residual_company_out = abs(out_residual_company)
+            rec.residual_out = abs(out_residual)
+            if (
+                float_is_zero(
+                    rec.residual_in,
+                    precision_rounding=rec.in_currency_id.rounding) and
+                float_is_zero(rec.residual_out,
+                              precision_rounding=rec.out_currency_id.rounding)
+            ):
+                rec.reconciled = True
+            else:
+                rec.reconciled = False
+
+    @api.multi
+    @api.depends("move_id.line_ids.amount_residual")
+    def _compute_payments(self):
+        for rec in self:
+            payment_lines = set()
+            for line in rec.move_id.line_ids:
+                payment_lines.update(line.mapped(
+                    "matched_credit_ids.credit_move_id.id"))
+                payment_lines.update(line.mapped(
+                    "matched_debit_ids.debit_move_id.id"))
+            rec.payment_move_line_ids = self.env[
+                "account.move.line"].browse(list(payment_lines))
+
+    @api.onchange("amount_out", "amount_in")
+    def _onchange_amounts(self):
+        self.rate = \
+            self.amount_in / self.amount_out if self.amount_out else 0.0
+
+    @api.onchange("rate")
+    def _onchange_rate(self):
+        self.amount_in = self.amount_out * self.rate
+
+    def _prepare_account_move(self):
```

Comment:
decorator?

Reply:
api.multi is the default, right? but yeah maybe for consistency with the other methods in the file I should also explicitly set it for theese methods.

Reply:
@lreficent yes it is the default. Some developers do not add it because it is 'the default', but what if at some point Odoo changes the API and this behaviour? Then you will have random functions breaking because of this.
I know this is a bug if and it feels like neatpicking but after all it only takes a few seconds to add an `@api.multi` right? 😉 
Thanks for adding!

Reply:
Yes, that's a fair point. Of course no problem to change it! Thanks